lib/string_reverse.rb

```diff
@@ -2,5 +2,19 @@
 # Time complexity: ?
 # Space complexity: ?
 def string_reverse(my_string)
-  raise NotImplementedError
+  if my_string == nil
+    return nil
+  end
+  if my_string == ""
+    return ""
+  end
+  n = 1
+  str_length = my_string.length
+  (str_length / 2).times do |i|
+    temp = my_string[i]
+    my_string[i] = my_string[str_length - n]
+    my_string[str_length - n] = temp

```

Comment:
If you wanted, you could do this on one line:

```ruby
my_string[i], my_string[str_length - n] = my_string[str_length - n], my_string[i]
```

but it's just as valid to use a temporary variable.